docs/metasploit-framework.wiki/Attacking-ADCS-ESC-Vulnerabilities.md

```diff
@@ -0,0 +1,916 @@
+# Workflow
+```mermaid
+flowchart TD
+	subgraph Legend
+		NODE1[/Setup directions\]
+		NODE2{{Vulnerabilities}}
+		NODE3[Modules]
+		NODE4{Desired Outcomes}
+	end
+	setup[/How to set up an ESC target\] --> esc[ESC Finder Module]
+	esc[ESC Finder Module] --> escexp{Find vulnerable certificate templates}
+	escexp{Find vulnerable certificate templates} --> icpr[Issue certificates via icpr_cert]
+	icpr[Issue certificates via icpr_cert] --> ESC1{{ESC1}}
+	ESC1{{ESC1}} -- Via PKINIT --> pkinit{Authenticate to Kerberos}
+	icpr[Issue certificates via icpr_cert] --> users[Request certificates on behalf of other users]
+	users[Request certificates on behalf of other users] --> ESC2{{ESC2}}
+	users[Request certificates on behalf of other users] --> ESC3{{ESC3}}
+	ESC2{{ESC2}} -- Via PKINIT --> pkinit{Authenticate to Kerberos}
+	ESC3{{ESC3}} -- Via PKINIT --> pkinit{Authenticate to Kerberos}
+```
```

Comment:
As someone seeing this for the first time, I'm not sure if the graph format is adding any additional clarity here 🤔 

I wonder if we should convert it to a table format or similar.

Or potentially simplify this to be:

```mermaid
flowchart TD
	escexp[Find vulnerable certificate templates\nvia ldap_esc_vulnerable_cert_finder] --> icpr[Issue certificates via icpr_cert]
	icpr[Issue certificates via icpr_cert] --> ESC1{{ESC1}}
	ESC1{{ESC1}} -- Via PKINIT --> pkinit{Authenticate to Kerberos}
	icpr[Issue certificates via icpr_cert] --> users[Request certificates on behalf of other users]
	users[Request certificates on behalf of other users] --> ESC2{{ESC2}}
	users[Request certificates on behalf of other users] --> ESC3{{ESC3}}
	ESC2{{ESC2}} -- Via PKINIT --> pkinit[Authenticate to Kerberos]
	ESC3{{ESC3}} -- Via PKINIT --> pkinit[Authenticate to Kerberos]
```

Reply:
So I think the idea here was to give a high level overview of some of the steps involved before diving into more specifics. The diagram tried to showcase that ESC1 has less steps than ESC2 and ESC3 to complete and that ESC2 and ESC3 are different in that you request certificates on behalf of other users vs requesting them for your own use.

This may tie into your point below though.

Reply:
Just cross-referencing https://github.com/rapid7/metasploit-framework/pull/17484/files#diff-1f6d911b96b743a9d2aa43759207202bc88ff8059e9287647964a81b4140b8e6 which adds placeholders for an AD CS overview which we'd be able to define in one place

Reply:
Sounds good, I can add in some definitions so long to this PR since I wrote up some explanations for myself in my Obsidian notes. Can probably use those to craft some definitions for these three attacks plus can provide a few others. Can't say I've gone through all of them yet though, but all of the ones we presently have support for I have notes on.

Reply:
That would be good, would you prefer I update things here and then we can move over to that file once the graph is a bit more polished and I can add a reference here linking to that?

Reply:
That sounds good to me :+1:

Reply:
After further review think I'll go with the suggestion you mentioned. I was trying to do too much in one graph with Mermaid.js which was making it look a bit crazy 😄

Reply:
Applied the suggested graph in 93f2e472bcdd53127e00e3ed0ec0fd4711799334